src/components/App/index.js

```diff
@@ -1,30 +1,50 @@
-import React, {Component} from 'react';
+import React, { Component } from 'react';
 import { connect } from 'react-redux';
+
+// through this react-redux bridge this component gets promoted to a container which is a "smart component"
+// that is a componnet that has direct connection to Redux
 import { today } from 'constants/apiQueries';
 
 import { fetchEpisodes } from 'redux/actions/episodesActions';
 
 import EpisodesList from 'components/App/EpisodesList';
 import Header from 'components/App/Header';
+import Pagination from 'components/App/Pagination';
 import Spinner from 'components/App/Spinner';
 
 import styles from './App.module.scss';
 
 class App extends Component {
+  state = {
+    currentPage: 1,
+    episodesPerPage: 12
+  }
+
   componentDidMount() {
     this.props.dispatch(fetchEpisodes());
   }
 
+  handlePageChange = event => {
+    this.setState({ currentPage: Number(event.target.id)});
+  }
+
   render() {
     const { error, loading, episodes } = this.props;
+    const { currentPage, episodesPerPage } = this.state;
+
+    const indexOfLastEpisode = currentPage * episodesPerPage;
```

Comment:
consider to hold everything for pagination in the store (current page number key) and then show only first 12, second 12 (...) filtered from the store data

Also you can consider keep current page in the url, what if I want to show someone third page and I'd love to have direct link? Pagination concept is not so easy, you need to think about cases and usage first. But 👍 for this basic implementation.